### PR TITLE
Refactor and update Apps session expiration

### PR DESCRIPTION
**Pull request by Aringoaway**
### Changes
1. Сome up with a text that makes more sense (old version: Your session has expired)
2. The disabled sidebar after the session has expired
3. Remove the button for closing in alert
4. Added function for word **here** to reload the page
- [x] Disabled buttons on the screen
- [x] Add locales for demo

https://user-images.githubusercontent.com/65866093/231772400-eeadb1db-cde5-4f71-8a92-e9fd9dcdf457.mp4

How it looks in the BitSwan in the Library:

image

### CODE REVIEW

File: src/containers/Alerts/index.js [hunk 1]
```diff
@@ -49,7 +50,7 @@ function AlertsComponent(props) {
 						className="shadow alerts-style"
 						fade={true}
 						isOpen={!alert.acked}
-						toggle={() => store.dispatch({ type: ACK_ALERT, key: alert.key })}
+						toggle={!sessionExpiration ? () => store.dispatch({ type: ACK_ALERT, key: alert.key }) : null}
```
Comment (Pe5h4) [inline src/containers/Alerts/index.js:53]: @Aringoaway PLease add a comment that your change will remove the close button for alert with expiration

File: src/containers/Sidebar/index.js [hunk 2]
```diff
@@ -80,66 +80,59 @@ const Sidebar = (props) => {
...
-				<div style={{display: "inline-block"}}>
-					<NavbarBrand {...props} isSidebarMinimized/>
+			<div className={`app-sidebar${isSidebarCollapsed ? " collapsed" : ""}`}>
+				{windowDimensions.width > 768 &&
+					<div style={{display: "inline-block"}}>
+						<NavbarBrand {...props} isSidebarMinimized/>
+					</div>
+				}
+				<div className="sidebar-nav">
+
```
Comment (Pe5h4) [inline src/containers/Sidebar/index.js:120]: @Aringoaway Remove empty line here

File: src/modules/auth/index.js [hunk 3]
```diff
@@ -72,6 +72,9 @@ export default class AuthModule extends Module {
 			if (this.OAuthToken != null) {
 				// Update the user info
 				let result = await this.updateUserInfo();
+				// if (this.App.Store != null) {
```
Comment (Pe5h4) [inline src/modules/auth/index.js:75]: @Aringoaway Why is this added and commented here?

**Review by Pe5h4 — CHANGES_REQUESTED** [review]

File: src/containers/Alerts/index.js [hunk 4]
```diff
@@ -59,10 +61,4 @@ function AlertsComponent(props) {
...
 }
 
-const mapStateToProps = state => {
-	return {
-		alerts: state.alerts.alerts
-	};
-}
-
-export default connect(mapStateToProps)(AlertsComponent);
+export default AlertsComponent;
```
Comment (Pe5h4) [inline src/containers/Alerts/index.js:64]: @Aringoaway Remove this line and use export default here: https://github.com/TeskaLabs/asab-webui/blob/8580edee187aaa26f64481b63b9cabfac7a5164f/src/containers/Alerts/index.js#L10

File: src/containers/Sidebar/index.js [hunk 5]
```diff
@@ -8,15 +8,17 @@ import SidebarBottomItem from './SidebarBottomItem';
 import {COLLAPSE_SIDEBAR} from "../../actions";
 
 
-const Sidebar = (props) => {
+export default function Sidebar (props) {
```
Comment (Pe5h4) [inline src/containers/Sidebar/index.js:11]: @Aringoaway Why did you make a function out of it? Just a question, I dont say its wrong

File: src/modules/auth/reducer.js [hunk 6]
```diff
@@ -5,6 +5,7 @@ const initialState = {
 	resources: [],
 	unauthorizedNavItem: [],
 	unauthorizedNavChildren: [],
+	sessionExpiration: false
```
Comment (Pe5h4) [inline src/modules/auth/reducer.js:8]: @Aringoaway For sake of the action naming, I propose to rename it to `sessionExpired`

File: src/modules/auth/reducer.js [hunk 7]
```diff
@@ -33,6 +34,12 @@ export default function reducer(state = initialState, action) {
 				unauthorizedNavChildren: action.unauthorizedNavChildren
 			}
 
+		case types.AUTH_SESSION_EXPIRATION:
+			return {
+				...state,
+				sessionExpiration: action.sessionExpiration
```
Comment (Pe5h4) [inline src/modules/auth/reducer.js:40]: @Aringoaway here it should be also `sessionExpired: action.sessionExpired`

**Review by Pe5h4 — CHANGES_REQUESTED** [review]

**Review by Pe5h4 — APPROVED** [review]
@Aringoaway I suppose that alert indicating ended  session due to inactivity is being removed when user is logged out

**Comment by Aringoaway** [comment]
@Pe5h4 Yes, sure the alert is removed

https://user-images.githubusercontent.com/65866093/229063607-6f55c4e6-e8f5-4e8d-8365-978d0202d6e9.mp4

**Comment by Aringoaway** [comment]
https://user-images.githubusercontent.com/65866093/229734167-2f03997c-2c40-48b6-a0b0-70884c2ff944.mp4

**Comment by Aringoaway** [comment]
https://user-images.githubusercontent.com/65866093/229735784-e7cf99d0-1d7c-4310-ab54-cb6e84927df6.mp4

File: src/containers/Alerts/index.js [hunk 8]
```diff
@@ -41,28 +43,32 @@ function AlertsComponent(props) {
...
+							<a
+								className="alert-link"
+								onClick={() => window.location.reload(false)} // refreshes the page
+								href="#"
+								rel="noreferrer"
+								target="_blank"
+							>
+								{t("Alerts|here")}
+							</a>
+						}
```
Comment (Aringoaway) [inline src/containers/Alerts/index.js:68]: @Pe5h4 I'm not sure about these changes. Do you think it's right or not?

https://user-images.githubusercontent.com/65866093/229820535-fec0b2d9-5c75-49b2-8b20-e3af67a002a3.mp4

File: src/containers/Alerts/index.js [hunk 9]
```diff
@@ -41,28 +43,32 @@ function AlertsComponent(props) {
...
 						isOpen={!alert.acked}
-						toggle={() => store.dispatch({ type: ACK_ALERT, key: alert.key })}
+						toggle={!sessionExpired ? () => store.dispatch({ type: ACK_ALERT, key: alert.key }) : null} // remove the close button for alert with expiration
 					>
 						{alert.shouldBeTranslated ? t(alert.message) : alert.message}
+						{" "}
+						{sessionExpired &&
+							<a
+								className="alert-link"
+								onClick={() => window.location.reload(false)} // refreshes the page
```
Comment (Pe5h4) [inline src/containers/Alerts/index.js:61]: @Aringoaway You dont need to add `false` prop to `window.location.reload()` - it has no use

File: src/containers/Alerts/index.js [hunk 10]
```diff
@@ -41,28 +43,32 @@ function AlertsComponent(props) {
...
+								href="#"
+								rel="noreferrer"
+								target="_blank"
```
Comment (Pe5h4) [inline src/containers/Alerts/index.js:64]: @Aringoaway Dont add `target`.  You dont need to have it here

File: src/containers/Alerts/index.js (hunk 8 above)
Comment (Pe5h4) [reply src/containers/Alerts/index.js:68]: @Aringoaway You know what? You dont need to use `a` tag at all. Why cant you use just
```
{sessionExpired &&
	<span onClick={window.location.reload} className="alert-span">
		{` ${t("Alerts|here")}`}
	</span>
}
```
and style the span with cursor and maybe some other styles?

Your approach seems to me bit like a overkill :))

**Review by Pe5h4 — CHANGES_REQUESTED** [review]

File: demo/public/locales/cs/translation.json [hunk 11]
```diff
@@ -91,7 +91,7 @@
 	"ASABAuthModule": {
 		"You are not authorized to use this application": "Nejste autorizováni k používání této aplikace",
 		"You are in DEV mode and using MOCK login parameters": "Jste v DEV režimu a používáte MOCK přihlašovací údaje",
-		"Your session has expired": "Vaše relace vypršela",
+		"You have been logged out due to inactivity. Please continue": "Byl jste odhlašen z důvodu nečinnosti. Pokračujte prosím",
```
Comment (ateska) [inline demo/public/locales/cs/translation.json:94]: Please continue to login.

**Review by ateska — CHANGES_REQUESTED** [review]

**Comment by Aringoaway** [comment]
disabled buttons

https://user-images.githubusercontent.com/65866093/231676895-a295d217-bde8-4c11-a608-609cd5433eb5.mp4

File: demo/public/locales/cs/translation.json (hunk 11 above)
Comment (ateska) [reply demo/public/locales/cs/translation.json:94]: Ok ... "+" operation for localization is a bit tricky.

File: src/modules/auth/index.js [hunk 12]
```diff
@@ -267,7 +267,14 @@ export default class AuthModule extends Module {
 		// if session has expired
 		if (!isUserInfoUpdated && oldUserInfo) {
 			oldUserInfo = null;
-			that.App.addAlert("danger", "ASABAuthModule|Your session has expired", 3600 * 1000, true);
+			that.App.addAlert("danger", "ASABAuthModule|You have been logged out due to inactivity.", 3600 * 1000, true);
+			if (that.App.Store != null) {
+				that.App.Store.dispatch({ type: types.AUTH_SESSION_EXPIRATION, sessionExpired: true });
+			}
+			[...document.getElementsByClassName('btn')].forEach(i => {
```
Comment (Pe5h4) [inline src/modules/auth/index.js:274]: @Aringoaway Write a comment that this disable buttons in the whole screen

**Review by Pe5h4 — CHANGES_REQUESTED** [review]
Just one comment, otherwise it looks great

**Review by ateska — APPROVED** [review]
Good job.

**Review by Pe5h4 — CHANGES_REQUESTED** [review]
@Aringoaway Update czech comments please, otherwise gut

**Review by Pe5h4 — APPROVED** [review]

**Comment by Aringoaway** [comment]
@Pe5h4 can you please see the latest changes added? I have added a new style file (breadcrumbs.scss) for breadcrumbs. 
On the website it looks like this:
image

File: src/containers/Header/BreadcrumbsRouter/breadcrumbs.scss [hunk 13]
```diff
@@ -0,0 +1,5 @@
+.breadcrumbs-disabled {
+	pointer-events: none;
+	cursor: default;
+	opacity: .65;
+}
```
Comment (Pe5h4) [inline src/containers/Header/BreadcrumbsRouter/breadcrumbs.scss:5]: @Aringoaway Add a newline at the end of the file

**Review by Pe5h4 — COMMENTED** [review]
@Aringoaway Its good, just dont forget to add a newline at the end of the files. That is the "unwritten" rule

File: src/modules/auth/index.js [hunk 14]
```diff
@@ -267,7 +267,19 @@ export default class AuthModule extends Module {
 		// if session has expired
 		if (!isUserInfoUpdated && oldUserInfo) {
 			oldUserInfo = null;
-			that.App.addAlert("danger", "ASABAuthModule|Your session has expired", 3600 * 1000, true);
+			that.App.addAlert("danger", "ASABAuthModule|You have been logged out due to inactivity.", 3600 * 1000, true);
+			if (that.App.Store != null) {
```
Comment (Pe5h4) [inline src/modules/auth/index.js:271]: @Aringoaway Shouldnt be the queryselectors wrapped in this condition? AS per my knowledge from your code, some of the features will not be disabled is (for any reason) the Store wont be available

**Comment by Aringoaway** [comment]
> Is it possible to add "something clickable" to the warning? For example underscore text or a button so that the user is prompted very specifically.

I have two examples of how this might look:
image
image

**Comment by ateska** [comment]
The second looks good to me.
But in this case, only "Continue to login" is a better text.

**Comment by Pe5h4** [comment]
@Aringoaway The second is more clear, that the user is expected to do the action to login redirection. I would think maybe of diminishing the padding around the text of the button.

File: src/containers/Alerts/alerts.scss [hunk 15]
```diff
@@ -4,4 +4,17 @@
 			color: inherit !important;
 		}
 	}
+	.alert-button {
```
Comment (Pe5h4) [inline src/containers/Alerts/alerts.scss:7]: @Aringoaway Do you really need this styling for button? I am not sure...

**Review by Pe5h4 — CHANGES_REQUESTED** [review]

**Comment by Pe5h4** [comment]
@Aringoaway Another remark is the text-color of the `Continue to Login` button in dark mode. It does not looks good. I think in this (and only this) case should stay white.

File: src/containers/Alerts/alerts.scss (hunk 15 above)
Comment (Aringoaway) [reply src/containers/Alerts/alerts.scss:9]: @Pe5h4 I need this styles, because of i assign button a color and other styles like margin, font-weight ... How do I do it without a class?

**Comment by Pe5h4** [comment]
> > @Aringoaway Another remark is the text-color of the `Continue to Login` button in dark mode. It does not looks good. I think in this (and only this) case should stay white.
> 
> image image

It looks better (but it is just my subjective point of view)

File: src/containers/Alerts/alerts.scss (hunk 15 above)
Comment (Pe5h4) [reply src/containers/Alerts/alerts.scss:9]: But why do you need font weight in the button? I thought some is already implemented by default. The same with the color (except the color of the text). You definitely dont need this styling

```
font-weight: bold;
cursor: pointer;
margin-left: 20px;
```

As it was used for the previous version of the implementation (<a> tag or <Link>)

**Comment by Aringoaway** [comment]
@ateska can you please tell us your opinion, is it better to have a button with dark colour or white in dark-mode? ^^^

**Review by Pe5h4 — APPROVED** [review]
